src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -52,10 +53,12 @@
    *
    * @param filePath blob file full path
    * @param chunkData uncompressed chunk data
-   * @return compressed chunk data
+   * @param blockSizeToAlignTo block size to align to for encryption
+   * @return padded compressed chunk data with actual length
```

Comment:
could you explain here why we need to return a pair? It may not be clear that the size returned is before padding, but the stream is after padding